### PR TITLE
hover: remove spacing between type definition 

### PR DESCRIPTION
**Pull request by aspeddro**
**Before:**

Neovim LSP builtin:
image

VSCode:
image

**After:**

Neovim:
image

VSCode:
image

### CODE REVIEW

**Comment by zth** [comment]
Having a new line there is intentional for VSCode, but I do see the problem in neovim. We'll probably need to treat them differently.

**Comment by aspeddro** [comment]
Fixed by 502d59124ca7338315d1f8b81f1f5969b85b3f7c

VSCode:

image

Neovim:

image

File: analysis/src/Protocol.ml [hunk 1]
```diff
@@ -97,7 +97,7 @@ let stringifyCompletionItem c =
     | None -> null
     | Some doc -> stringifyMarkupContent doc)
 
-let stringifyHover s = Printf.sprintf {|{"contents": "%s"}|} (Json.escape s)
+let stringifyHover value = Printf.sprintf {|{"contents": %s}|} (stringifyMarkupContent {kind = "markdown"; value})
```
Comment (aspeddro) [inline analysis/src/Protocol.ml:100]: Use [MarkupContent](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markupContentInnerDefinition). [MarkedString](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markedString) and [MarkedString[]](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#markedString) are depreacted.

**Comment by zth** [comment]
That looks good. What about the dividers, would you like to test removing them completely in vim too?

**Comment by aspeddro** [comment]
Neovim automatically adds [dividers between code blocks](https://github.com/neovim/neovim/blob/fae754073289566051433fae74ec65783f9e7a6a/runtime/lua/vim/lsp/util.lua#L1224-L1243) even though it doesn't return dividers `\n---\n`


```lua
result = {
      contents = {
        kind = "markdown",
        value = "```rescript\n(\n  Belt.Array.t<Layer.t>,\n  Layer.t => string,\n) => Belt.Array.t<string>\n```\n```rescript\ntype Belt.Array.t<'a> = array<'a>\n```\n\n```rescript\ntype Layer.t = Alface | Tomate
| Queijo | Cebola | Pao\n```\n\n\n\n  `map(xs, f)`\n\n  Returns a new array by calling `f` for each element of `xs` from the beginning to end.\n\n  ```res example\n  Belt.Array.map([1, 2], (x) => x + 1) == [3, 4]
\n  ```\n"
      }
}
```

image

**Comment by aspeddro** [comment]
Dividers do indeed improve readability.

File: analysis/src/Hover.ml [hunk 2]
```diff
@@ -100,12 +100,12 @@ let hoverWithExpandedTypes ~docstring ~file ~package ~supportsMarkdownLinks typ
                Markdown.goToDefinitionText ~env ~pos:loc.Warnings.loc_start
              else ""
            in
-           "\n" ^ Markdown.spacing
+           Markdown.divider ^ (if supportsMarkdownLinks then Markdown.spacing else "")
```
Comment (zth) [inline analysis/src/Hover.ml:103]: We should probably name `supportsMarkdownLinks` something smarter now that it handles more cases, but that's a question for later.

**Review by zth — APPROVED** [review]

**Comment by zth** [comment]
@aspeddro ready to merge as soon as the tests are green again 👍

**Comment by aspeddro** [comment]
I don't know why the tests are failing, all the changes are committed.

**Comment by zth** [comment]
Can you resolve the merge conflict here too? I'll have a look at the tests tomorrow.

**Comment by aspeddro** [comment]
Done

**Pull request merged**